code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -264,3 +264,278 @@
+	M.remove_status_effect(STATUS_EFFECT_KATANA_CURSE)
+	. = ..()
+
+
+#define LEFT_SLASH "Harm"
+#define RIGHT_SLASH "Disarm"
+#define COMBO_STEPS "steps"
+#define COMBO_PROC "proc"
+#define ATTACK_STRIKE "Hilt Strike"
+#define ATTACK_SLICE "Wide Slice"
+#define ATTACK_DASH "Dash Attack"
+#define ATTACK_CUT "Tendon Cut"
+#define ATTACK_HEAL "Dark Heal"
+#define ATTACK_SHATTER "Shatter"
+
+
+/obj/item/cursed_katana
+	name = "cursed katana"
+	desc = "A katana used to seal something vile away long ago. \
+	Even with the weapon destroyed, all the pieces containing the creature have coagulated back together to find a new host."
+	icon = 'icons/obj/lavaland/artefacts.dmi'
+	icon_state = "cursed_katana"
+	lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items_righthand.dmi'
+	force = 15
+	armour_penetration_percentage = 40
+	armour_penetration_flat = 10
+	block_chance = 50
+	sharp = TRUE
+	w_class = WEIGHT_CLASS_HUGE
+	attack_verb = list("attack", "slash", "stab", "slice", "tear", "lacerate", "rip", "dice", "cut")
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	var/shattered = FALSE
+	var/drew_blood = FALSE
+	var/timerid
+	var/list/input_list = list()
+	var/list/combo_strings = list()
+	var/static/list/combo_list = list(
+		ATTACK_STRIKE = list(COMBO_STEPS = list(LEFT_SLASH, LEFT_SLASH, RIGHT_SLASH), COMBO_PROC = .proc/strike),
+		ATTACK_SLICE = list(COMBO_STEPS = list(RIGHT_SLASH, LEFT_SLASH, LEFT_SLASH), COMBO_PROC = .proc/slice),
+		ATTACK_DASH = list(COMBO_STEPS = list(LEFT_SLASH, RIGHT_SLASH, RIGHT_SLASH), COMBO_PROC = .proc/dash),
+		ATTACK_CUT = list(COMBO_STEPS = list(RIGHT_SLASH, RIGHT_SLASH, LEFT_SLASH), COMBO_PROC = .proc/cut),
+		ATTACK_HEAL = list(COMBO_STEPS = list(LEFT_SLASH, RIGHT_SLASH, LEFT_SLASH, RIGHT_SLASH), COMBO_PROC = .proc/heal),
+		ATTACK_SHATTER = list(COMBO_STEPS = list(RIGHT_SLASH, LEFT_SLASH, RIGHT_SLASH, LEFT_SLASH), COMBO_PROC = .proc/shatter),
+		)
+
+/obj/item/cursed_katana/Initialize(mapload)
+	. = ..()
+	for(var/combo in combo_list)
+		var/list/combo_specifics = combo_list[combo]
+		var/step_string = english_list(combo_specifics[COMBO_STEPS])
+		combo_strings += ("<span class='notice'><b>[combo]</b> - [step_string]</span>")
+
+/obj/item/cursed_katana/examine(mob/user)
+	. = ..()
+	. += drew_blood ? ("<span class='notice'>It's sated... for now.</span>") : ("<span class='danger'>It will not be sated until it tastes blood.</span>")
+	. += combo_strings
+
+/obj/item/cursed_katana/dropped(mob/user)
+	. = ..()
+	reset_inputs(null, TRUE)
+
+/obj/item/cursed_katana/attack_self(mob/user)
+	. = ..()
+	reset_inputs(user, TRUE)
+
+/obj/item/cursed_katana/attack(mob/living/target, mob/user, click_parameters)
+	if(target.stat == DEAD || target == user) //No, you can not stab yourself to cloak / not take the penalty for not drawing blood
+		return ..()
+	if(HAS_TRAIT(user, TRAIT_PACIFISM))
+		to_chat(user, "<span class='warning'>You don't want to harm [target]!</span>")
+		return
```

Comment:
I think you want to return TRUE here to stop the attack chain
```suggestion
		return TRUE
```

Reply:
Chain stops without true

Reply:
TRUE ensures after attack isnt called

Reply:
done